### DISCUSSION TITLE
Lost data of alpha channel when -negate

### DISCUSSION THREAD

**Discussion by kelliray90** [ImageMagick/ImageMagick#6482] [Help] [answered]
### ImageMagick version

7.1.0-58 Q16-HDR

### Operating system, version and so on

windows 10

### Description

I'm trying to delete specific area according mask by below command:
` magick base.png ( mask.png -negate ) -alpha off -compose copy-opacity -define compose:clamp=off -composite result.png`
and the result is lost alpha data of base image.

I also try this command but it's not work:
`magick base.png mask.png -channel A -combine result.png`

Download all images here:  [lost-alpha.zip](https://github.com/ImageMagick/ImageMagick/files/12034772/lost-alpha.zip)
Base image (the white circle is transparent):
![base](https://github.com/ImageMagick/ImageMagick/assets/28614684/2de6b448-1865-4f62-b038-654b1f901ea3)
Mask image:
![mask](https://github.com/ImageMagick/ImageMagick/assets/28614684/6a22a90b-7b65-4f73-bc57-68e971b9a05e)

The result (lost transparent of base image):
![result](https://github.com/ImageMagick/ImageMagick/assets/28614684/abfc8f6b-e017-4647-8798-4d3bacc19ff3)
The expected result (keep transparent of both images): 
![expected result](https://github.com/ImageMagick/ImageMagick/assets/28614684/f33c3fca-64f6-44ee-b598-1becef676a0a)

**Comment by snibgo** [comment]
`mask.png` has an alpha channel, so `-negate` will negate alpha in addition to the colour channels. To restrict the operation to just the color channels, assuming they are RGB, then:
```
 magick base.png ( mask.png -channel RGB -negate +channel ) -alpha off -compose copy-opacity -define compose:clamp=off -composite result.png
```
But your "`-alpha off`" implies you want only the colour channels. So you could:
```
 magick base.png ( mask.png -alpha off -negate ) -alpha off -compose copy-opacity -define compose:clamp=off -composite result.png
```

**Reply by kelliray90** [reply]
Thanks for fast reply! I tried your command and it still not work. The result that I want is keep RGB as is, the circle and lines must be transparent.
In the current result image, only lines are transparent, the circle is not transparent.

**Comment by snibgo** [comment, marked as answer]
Yes, sorry, I didn't look carefully at what you wanted to achieve. (This was partly because I use a dark theme, so your "white circle" is black on my screen.)

CopyOpacity copies the opacity of the mask to the base image, *replacing* the opacity of the base image. But you want to *combine* the opacity from the two images. To combine, I suggest using "`-compose Darken -composite`" so the output opacity will be the lower of the two inputs. We use "`-alpha extract`" to extract the alpha channel from the input, and finally "`-compose CopyOpacity -composite`" to copy the combined opacity to the output image.

Windows syntax:
```
magick ^
  kellBase.png ^
  ( +clone -alpha extract ^
    ( kellMask.png -alpha off -negate ) ^
    -compose Darken -composite ^
  ) ^
  -alpha off -compose CopyOpacity -composite ^
  out.png
```

**Reply by kelliray90** [reply]
It works like a charm! Thank you so much for the detail explanation.
